test/Correlate.AspNet.Tests/CorrelateFeatureNet48Tests.cs

```diff
@@ -0,0 +1,325 @@
+        ICorrelationIdFactory correlationIdFactory = Substitute.For<ICorrelationIdFactory>();
+        IActivityFactory activityFactory = Substitute.For<IActivityFactory>();
+        IOptions<CorrelateOptionsNet48> options = Substitute.For<IOptions<CorrelateOptionsNet48>>();
+
+        yield return [null, correlationIdFactory, activityFactory, options, nameof(loggerFactory), typeof(ArgumentNullException)];
+        yield return [loggerFactory, null, activityFactory, options, nameof(correlationIdFactory), typeof(ArgumentNullException)];
+        yield return [loggerFactory, correlationIdFactory, null, options, nameof(activityFactory), typeof(ArgumentNullException)];
+        yield return [loggerFactory, correlationIdFactory, activityFactory, null, nameof(options), typeof(ArgumentException)];
+    }
+
+    [Fact]
+    public void Given_that_activity_in_items_was_replaced_with_something_else_when_correlating_has_stopped_it_should_not_throw()
+    {
+        _sut.StartCorrelating(_httpContext);
+        _httpContext.Items.Keys.Cast<string>().Should().Contain(CorrelateFeatureNet48.RequestActivityKey);
+        _httpContext.Items[CorrelateFeatureNet48.RequestActivityKey] = new object();
+
+        // Act
+        Action act = () => _sut.StopCorrelating(_httpContext);
+
+        // Assert
+        act.Should().NotThrow();
+        _activityFactoryMock.Received(1).CreateActivity();
+        _activityMock.Received(1).Start(CorrelationId);
+        _activityMock.DidNotReceive().Stop();
+    }
+
+    [Fact]
+    public void Given_that_activity_is_not_in_items_when_correlating_has_stopped_it_should_not_throw()
+    {
+        _sut.StartCorrelating(_httpContext);
+        _httpContext.Items.Keys.Cast<string>().Should().Contain(CorrelateFeatureNet48.RequestActivityKey);
+        _httpContext.Items.Clear();
+
+        // Act
+        Action act = () => _sut.StopCorrelating(_httpContext);
+
+        // Assert
+        act.Should().NotThrow();
+        _activityFactoryMock.Received(1).CreateActivity();
+        _activityMock.Received(1).Start(CorrelationId);
+        _activityMock.DidNotReceive().Stop();
+    }
+
+    [Fact]
+    public void Given_that_correlating_has_not_started_when_firing_to_send_headers_it_should_not_add_correlationId_header_to_response()
+    {
+        _options.IncludeInResponse.Should().BeTrue();
+        _options.RequestHeaders.Should().NotBeNullOrEmpty();
+
+        // Act
+
+        // Assert
+        _httpContext.Response.Headers.Keys.Cast<string>().Should().BeEmpty();
+    }
+
+    [Fact]
+    public void Given_that_response_already_contains_correlation_header_when_firing_to_send_headers_it_should_not_overwrite_the_correlationId_header()
+    {
+        _options.IncludeInResponse.Should().BeTrue();
+        _options.RequestHeaders.Should().NotBeNullOrEmpty();
+
+        const string existingCorrelationId = "existing-id";
+        _httpContext.Request.Headers.Add(_options.RequestHeaders![0], existingCorrelationId);
```

Comment:
For this test, the arrange should be to add the header to the response, not the request. The purpose of this test is to assert that the response header is not added/overwritten if some code further down the request pipeline already added the header (eg. in a action or controller).

It also means you can revert the changed assertion on L236 back to what it was.